Kernel/VirtIO/VirtIOQueue.cpp

```diff
@@ -22,9 +22,9 @@ VirtIOQueue::VirtIOQueue(u16 queue_size, u16 notify_offset)
     // TODO: ensure alignment!!!
     u8* ptr = m_queue_region->vaddr().as_ptr();
     memset(ptr, 0, m_queue_region->size());
-    m_descriptors = reinterpret_cast<VirtIOQueueDescriptor*>(ptr);
-    m_driver = reinterpret_cast<VirtIOQueueDriver*>(ptr + size_of_descriptors);
-    m_device = reinterpret_cast<VirtIOQueueDevice*>(ptr + size_of_descriptors + size_of_driver);
+    m_descriptors = adopt_own_if_nonnull(reinterpret_cast<VirtIOQueueDescriptor*>(ptr));
```

Comment:
Hmmm. I don't think it's correct to store these in OwnPtrs at all. The pointers here point into a manually allocated kernel region, rather than something allocated with `kmalloc`.

Reply:
This PR does not introduce any behaviour changes, it only makes the existing behaviour more explicit.
That said, I agree, but I'm not familiar with what this is doing, so I'll leave that change to @IdanHo.

Reply:
I'm gonna merge this, we can improve the hidden weirdness this makes clear in a later PR.